stdlib.sh

```diff
@@ -26,6 +25,89 @@ direnv_config_dir=${XDG_CONFIG_DIR:-$HOME/.config}/direnv
 # algorithm and so it won't be re-exported.
 export DIRENV_IN_ENVRC=1
 
+__env_strictness() {
+  local mode tempfile old_strictness
+  local -i res
+
+  res=0
+  tempfile=$(mktemp)
+  mode="$1"
+  shift
+
+  set +o | grep 'pipefail\|nounset\|errexit' >"$tempfile"
+  old_strictness=$(<"$tempfile")
+  rm -f "$tempfile"
+
+  case "$mode" in
+  strict)
+    set -o errexit -o nounset -o pipefail
+    ;;
+  unstrict)
+    set +o errexit +o nounset +o pipefail
+    ;;
+  *)
+    echo "Unknown strictness mode '${mode}'."
+    exit 1
+    ;;
+  esac
+
+  if (($#)); then
```

Comment:
I _could_ remove this particular test because of the early exit in `strict_env` and `unstrict_env`, but this still seems safer.